webapp/components/wpt-bsf.js

```diff
@@ -0,0 +1,215 @@
+        }
+        paper-button {
+          color: black;
+          text-transform: none;
+        }
+      </style>
+      <div class="bsf">
+        <div class="left">
+          <h5>Channel tabs</h5>
+          <div class="channel">
+            <paper-button class\$="[[stableButtonClass(isExperimental)]]" onclick="[[clickStable]]">Stable</paper-button>
+            <paper-button class\$="[[experimentalButtonClass(isExperimental)]]" onclick="[[clickExperimental]]">Experimental</paper-button>
+          </div>
+          <h5>Last updated GitHub SHA</h5>
+          <div class="sha">
+            <iron-icon class="github-icon" src="/static/github.svg"></iron-icon>
+            <a class="link" href="[[githubHref]]" target="_blank"><paper-button>[[shortSHA]]</paper-button></a>
+          </div>
+          <h5>Click + drag on graphs to zoom, right click to un-zoom</h5>
+        </div>
+        <google-chart type="line" class="chart" data="[[data]]" options="[[chartOptions]]"></google-chart>
+      </div>
+    `;
+  }
+
+  static get is() {
+    return 'wpt-bsf';
+  }
+
+  static get properties() {
+    return {
+      data: Array,
+      sha: String,
+      shortSHA: {
+        type: String,
+        computed: 'computeShortSHA(sha)',
+      },
+      githubHref: {
+        type: String,
+        computed: 'computeGitHubHref(sha)',
+      },
+      isExperimental: {
+        type: Boolean,
+        value: false,
+      },
+      chartOptions: {
+        type: Object,
+        value: {
+          width: 800,
+          height: 350,
+          chartArea: {
+            height: '80%',
+          },
+          hAxis: {
+            title: 'Date',
+            format: 'MMM-YYYY',
+          },
+          vAxis: {
+            title: 'Tests that fail in exactly 1 browser',
+          },
+          explorer: {
+            actions: ['dragToZoom', 'rightClickToReset'],
+            axis: 'horizontal',
+            keepInBounds: true,
+            maxZoomIn: 4.0,
+          },
+          colors: ['#4285f4', '#ea4335', '#fbbc04'],
+        }
+      },
+    };
+  }
+
+  constructor() {
+    super();
+    this.clickStable = () => {
+      if (this.isExperimental === false) {
+        return;
+      }
+      this.isExperimental = false;
+      this.loadBSFData();
+    };
+    this.clickExperimental = () => {
+      if (this.isExperimental === true) {
+        return;
+      }
+      this.isExperimental = true;
+      this.loadBSFData();
+    };
+    this.loadBSFData();
+  }
+
+  computeGitHubHref(sha) {
+    return 'https://github.com/web-platform-tests/wpt/commit/' + sha;
+  }
+
+  computeShortSHA(sha) {
+    return sha.slice(0, 10);
+  }
+
+  stableButtonClass(isExperimental) {
+    return this.getChannelClass(!isExperimental);
+  }
+
+  experimentalButtonClass(isExperimental) {
+    return this.getChannelClass(isExperimental);
+  }
+
+  getChannelClass(isChannel) {
+    if (isChannel) {
+      return 'selected';
+    }
+    return 'unselected';
```

Comment:
```suggestion
    return isChannel ? 'selected' : 'unselected';
```

Also, I feel like 'isChannel' is not an understandable parameter name. I would be tempted to drop this method entirely and just inline the checks as:

```
  stableButtonClass(isExperimental) {
    return isExperimental ? 'unselected' : 'selected';
  }

  experimentalButtonClass(isExperimental) {
    return isExperimental ? 'selected' : 'unselected';
  }
```